pkg/types/azure/validation/platform_test.go

```diff
@@ -0,0 +1,58 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/openshift/installer/pkg/types/azure"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidatePlatform(t *testing.T) {
+	cases := []struct {
+		name     string
+		platform *azure.Platform
+		valid    bool
+	}{
+		{
+			name: "minimal",
+			platform: &azure.Platform{
+				Region: "eastus",
+			},
+			valid: true,
+		},
+		{
+			name: "invalid region",
+			platform: &azure.Platform{
+				Region: "bad-region",
+			},
+			valid: false,
+		},
+		{
+			name: "valid machine pool",
```

Comment:
valid and invalid pool look the same here?

Reply:
fixed the test, and added a TODO in the machinepool validation, which is not validating anything just yet. https://github.com/openshift/installer/pull/1460/commits/fbdbf5cc1e877b3a377367e44ddb9d333e68f169